developer_manual/digging_deeper/groupware/contacts_menu.rst

```diff
@@ -0,0 +1,89 @@
+=============
+Contacts Menu
+=============
+
+Nextcloud shows a *Contacts menu* in the right corner of the header. This menu lists a user's contacts. These contact entries can be extended by apps.
+
+Apps that extend the contacts menu implement an IProvider. The providers ``process`` method is called for every entry show in the contacts menu.
+
+.. code-block:: php
+    :caption: lib/ContactsMenu/MyProvider.php
+
+    <?php
+
+    namespace OCA\MyApp\ContactsMenu;
+
+    use OCP\Contacts\ContactsMenu\IEntry;
+    use OCP\Contacts\ContactsMenu\IProvider;
+
+    class MyProvider implements IProvider {
+        public function process(IEntry $entry): void {
+            // todo: something useful
+        }
+    }
+
+
+.. code-block:: xml
+    :caption: appinfo/info.xml
+    :emphasize-lines: 6-8
+
+    <?xml version="1.0"?>
+    <info xmlns:xsi= "http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="https://apps.nextcloud.com/schema/apps/info.xsd">
+        <id>my_app</id>
+        <name>My App</name>
+        <contactsmenu>
+            <provider>OCA\MyApp\ContactsMenu\LinkActionProvider</provider>
+        </contactsmenu>
+    </info>
+
+Accessing contact info
+^^^^^^^^^^^^^^^^^^^^^^
+
+The ``IEntry`` objects offer getters to contact infos:
+
+* ``getFullName()``: Get full name. Falls back to an empty string if no full name is set.
+* ``getEMailAddresses()``: Get all email addresses.
+* ``getAvatar()``: Get the avatar URI.
+* ``getProperty(string $name)``: Read a property of the contact. Return NULL if the property is not set.
```

Comment:
Are the properties defined somewhere? If yes, a link/cross-link would be nice.
Alternatively, one could add a table or similar to the documentation.

Reply:
This looks better. Thanks